### PR TITLE
Refactoring in the build script

### PR DESCRIPTION
**Pull request by psfinaki**
Polished a few things after recent changes, see comments as well. Tried all the targets locally, seem to work.

### CODE REVIEW

File: build.fsx [hunk 1]
```diff
@@ -6,10 +6,16 @@ open Fake.Core
 open Fake.Core.TargetOperators
 open Fake.IO
 
+type Browser =
```
Comment (psfinaki) [inline build.fsx:9]: To avoid passing bools around.

File: build.fsx [hunk 2]
```diff
@@ -52,10 +58,10 @@ let serverWatcher() =
 
 let clientWatcher openBrowser = 
     async {
-        // NOTE: space in flag is left intentionally (see https://github.com/psfinaki/CheckYourCzech/pull/743#discussion_r506929335)
-        let openFlag = if openBrowser then " --open" else ""
-        let webpackConfig = Path.combine clientPath "webpack.development.js"
-        let webpackCommand = sprintf "webpack-dev-server --config %s%s" webpackConfig openFlag
+        let webpackCommand = openBrowser |> function
+            | Open -> sprintf "webpack-dev-server --config %s --open" webpackDevConfig
+            | DontOpen -> sprintf "webpack-dev-server --config %s" webpackDevConfig
```
Comment (psfinaki) [inline build.fsx:63]: More safety and explicity

File: build.fsx [hunk 3]
```diff
@@ -98,22 +99,13 @@ Target.create "RunScraper" (fun _ ->
     runDotNet "run" scraperPath
 )
 
-Target.create "RunClient" (fun _ ->
```
Comment (psfinaki) [inline build.fsx:101]: This is just not used

File: build.fsx [hunk 4]
```diff
@@ -72,21 +78,16 @@ Target.create "InstallClient" (fun _ ->
     runDotNet "restore" clientPath
 )
 
-Target.create "RestoreServer" (fun _ ->
```
Comment (psfinaki) [inline build.fsx:75]: .NET Core [implicitly runs restore](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build#implicit-restore) prior to build

File: build.fsx [hunk 5]
```diff
@@ -98,22 +99,13 @@ Target.create "RunScraper" (fun _ ->
...
-    yarn webpackCommand
-)
-
 Target.create "RunE2ETests" (fun _ ->
-    runDotNet "build" clientTestsPath
     runDotNet "test" clientTestsPath
 )
 
 Target.create "RunWebWithE2ETests" (fun _ ->
-    runDotNet "build" clientTestsPath
```
Comment (psfinaki) [inline build.fsx:113]: .NET Core [implicitly runs](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test#description) build before tests

**Review by reflash — APPROVED** [review]

**Pull request merged**